sips/pending/_posts/2016-06-25-trailing-commas.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: sip
+disqus: true
+title: SIP-NN - Trailing Commas
+---
+
+# SIP-NN: Trailing Commas
+
+**By: Dale Wijnand**
+
+## History
+
+| Date          | Version       |
+|---------------|---------------|
+| Jun 25th 2016 | Initial Draft |
+
+## Motivation
+
+When using a comma-separated sequence of elements on multiple lines, such as:
+
+{% highlight scala %}
+Seq(
+  foo,
+  bar,
+  baz
+)
+{% endhighlight %}
+
+It is quite inconvenient to remove or comment out any element because one has to think about the fact that the last element mustn't have a trailing comma:
+
+{% highlight scala %}
+Map(
+  foo,
+  bar //,
+//  baz
+)
+{% endhighlight %}
+
+Secondly, it is quite inconvenient to re-order the sequence, for instance if you wanted `baz` before `bar` you need to micromanage which is followed by a comma and which isn't:
+
+{% highlight scala %}
+val xs = Seq(
+  foo,
+  baz   // This isn't going to work
+  bar,
+)
+{% endhighlight %}
+
+Additionally, this means that there is a lot of noise in diffs, such as:
+
+{% highlight diff %}
+@@ -4,7 +4,8 @@
+ Map(
+   foo,
+   bar,
+-  baz
++  baz,
++  quux
+ )
+{% endhighlight %}
+
+Also such feature would massively simplify generating Scala source code.
```

Comment:
s/massively/modestly/ IMO — "massively" just seems like hyperbole. or is it really that big an issue in your experience? (I do have some experience writing code generators that omitted trailing commas and found it annoying, but really not that a big deal, either.)